manifests/config.pp

```diff
@@ -14,36 +14,13 @@
   )
   create_resources('jenkins::sysconfig', $config_hash)
 
-  $dir_params = {
-    ensure => directory,
-    owner  => $::jenkins::user,
-    group  => $::jenkins::group,
-    mode   => '0755',
+  if (
+    ! defined(Jenkins::Sysconfig['JENKINS_HOME'])  and
+    ! has_key($::jenkins::config_hash, 'JENKINS_HOME')
```

Comment:
Not checking will make sure manifests fail when they have a JENKINS_HOME in the config hash or manually defined. I should probably document the use case for having a different localstatedir from the jenkins's user home dir that will require manage_user => false then.